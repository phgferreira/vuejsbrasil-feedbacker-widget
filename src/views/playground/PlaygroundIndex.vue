<template>
  <div class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-main">
    <h1 class="text-6xl font-black text-brand-gray">Playground</h1>
    <p class="text-2xl mt-3 font-regular text-brand-gray">Este Ã© o Playground, use para testar o Widget.</p>
  </div>
  <div class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray">
    <h1 class="text-6xl font-black text-brand-gray-dark">Coloque seu Widget bem aqui</h1>
    <widget-index></widget-index>
  </div>
</template>

<script setup lang="ts">

import WidgetIndex from '@/views/widget/WidgetIndex.vue'
</script>
