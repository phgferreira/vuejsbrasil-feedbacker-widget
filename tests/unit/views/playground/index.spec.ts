import { shallowMount } from '@vue/test-utils'
import PlaygroundIndex from '../../../../src/views/playground/PlaygroundIndex.vue'

describe('<Playground />', () => {
  it('deve renderizar o playground', () => {
    const wrapper = shallowMount(PlaygroundIndex)
    expect(wrapper).toMatchSnapshot()
  })
})
