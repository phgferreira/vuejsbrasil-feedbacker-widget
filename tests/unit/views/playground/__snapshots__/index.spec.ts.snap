// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playground /> deve renderizar o playground 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      
      <div
        class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-main"
      >
        <h1
          class="text-6xl font-black text-brand-gray"
        >
          Playground
        </h1>
        <p
          class="text-2xl mt-3 font-regular text-brand-gray"
        >
          Este Ã© o Playground, use para testar o Widget.
        </p>
      </div>
      <div
        class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray"
      >
        <h1
          class="text-6xl font-black text-brand-gray-dark"
        >
          Coloque seu Widget bem aqui
        </h1>
        <widget-stub />
      </div>
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {},
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {},
          "_value": Object {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {},
      },
      "refs": Object {
        "VTU_COMPONENT": Object {},
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {},
      },
      "slots": Object {},
      "slotsProxy": null,
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {},
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {},
          "da": null,
          "data": Object {},
          "devtoolsRawSetupState": Object {
            "Widget": Object {
              "__name": "WidgetIndex",
              "render": [Function],
              "setup": [Function],
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": Object {},
          "exposed": Object {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {},
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": Object {
            "attrs": Object {},
            "emit": [Function],
            "expose": [Function],
            "slots": Object {},
          },
          "setupState": Object {
            "Widget": Object {
              "__name": "WidgetIndex",
              "render": [Function],
              "setup": [Function],
            },
          },
          "slots": Object {},
          "slotsProxy": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": ,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "ce": undefined,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "ce": undefined,
                    "children": "Playground",
                    "component": null,
                    "ctx": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <h1
                      class="text-6xl font-black text-brand-gray"
                    >
                      Playground
                    </h1>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "class": "text-6xl font-black text-brand-gray",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "h1",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "ce": undefined,
                    "children": "Este Ã© o Playground, use para testar o Widget.",
                    "component": null,
                    "ctx": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <p
                      class="text-2xl mt-3 font-regular text-brand-gray"
                    >
                      Este Ã© o Playground, use para testar o Widget.
                    </p>,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {
                      "class": "text-2xl mt-3 font-regular text-brand-gray",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "p",
                  },
                ],
                "component": null,
                "ctx": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-main"
                >
                  <h1
                    class="text-6xl font-black text-brand-gray"
                  >
                    Playground
                  </h1>
                  <p
                    class="text-2xl mt-3 font-regular text-brand-gray"
                  >
                    Este Ã© o Playground, use para testar o Widget.
                  </p>
                </div>,
                "key": null,
                "patchFlag": -1,
                "props": Object {
                  "class": "w-full h-3/4 flex flex-col justify-center items-center bg-brand-main",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "ce": undefined,
                    "children": "Coloque seu Widget bem aqui",
                    "component": null,
                    "ctx": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <h1
                      class="text-6xl font-black text-brand-gray-dark"
                    >
                      Coloque seu Widget bem aqui
                    </h1>,
                    "key": null,
                    "patchFlag": -1,
                    "props": Object {
                      "class": "text-6xl font-black text-brand-gray-dark",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "h1",
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": Object {
                      "a": null,
                      "accessCache": Object {},
                      "appContext": Object {
                        "app": [Circular],
                        "components": Object {
                          "transition": Object {
                            "name": "transition",
                          },
                          "transition-group": Object {
                            "name": "transition-group",
                          },
                        },
                        "config": Object {
                          "compilerOptions": Object {},
                          "errorHandler": undefined,
                          "globalProperties": Object {},
                          "isNativeTag": [Function],
                          "optionMergeStrategies": Object {},
                          "performance": false,
                          "warnHandler": undefined,
                        },
                        "directives": Object {},
                        "emitsCache": WeakMap {},
                        "mixins": Array [
                          Object {
                            "beforeCreate": [Function],
                          },
                        ],
                        "optionsCache": WeakMap {},
                        "propsCache": WeakMap {},
                        "provides": Object {},
                        "reload": [Function],
                      },
                      "asyncDep": null,
                      "asyncResolved": false,
                      "attrs": Object {},
                      "attrsProxy": null,
                      "bc": null,
                      "bm": null,
                      "bu": null,
                      "bum": null,
                      "c": null,
                      "components": null,
                      "ctx": Object {},
                      "da": null,
                      "data": Object {},
                      "directives": null,
                      "ec": null,
                      "effect": ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": Array [
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                      "emit": [Function],
                      "emitsOptions": null,
                      "emitted": null,
                      "exposeProxy": null,
                      "exposed": null,
                      "inheritAttrs": undefined,
                      "isDeactivated": false,
                      "isMounted": true,
                      "isUnmounted": false,
                      "m": null,
                      "next": null,
                      "parent": [Circular],
                      "props": Object {},
                      "propsDefaults": Object {},
                      "propsOptions": Array [
                        Object {},
                        Array [],
                      ],
                      "provides": Object {},
                      "proxy": Object {},
                      "refs": Object {},
                      "render": [Function],
                      "renderCache": Array [],
                      "root": [Circular],
                      "rtc": null,
                      "rtg": null,
                      "scope": EffectScope {
                        "_active": true,
                        "cleanups": Array [],
                        "detached": true,
                        "effects": Array [
                          ReactiveEffect {
                            "active": true,
                            "allowRecurse": true,
                            "deps": Array [
                              Set {
                                [Circular],
                              },
                            ],
                            "fn": [Function],
                            "onTrack": undefined,
                            "onTrigger": undefined,
                            "parent": undefined,
                            "scheduler": [Function],
                          },
                        ],
                        "parent": undefined,
                      },
                      "setupContext": Object {
                        "attrs": Object {},
                        "emit": [Function],
                        "expose": [Function],
                        "slots": Object {},
                      },
                      "setupState": Object {},
                      "slots": Object {},
                      "slotsProxy": Object {},
                      "sp": null,
                      "subTree": Object {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": null,
                        "component": null,
                        "ctx": [Circular],
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": <widget-stub />,
                        "key": null,
                        "patchFlag": 0,
                        "props": Object {},
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 1,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": "widget-stub",
                      },
                      "suspense": null,
                      "suspenseId": 0,
                      "type": Object {
                        "model": undefined,
                        "name": "Widget",
                        "props": Object {},
                        "setup": [Function],
                      },
                      "u": null,
                      "uid": 2,
                      "um": null,
                      "update": [Function],
                      "vnode": [Circular],
                      "withProxy": null,
                    },
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <widget-stub />,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 4,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Object {
                      "model": undefined,
                      "name": "Widget",
                      "props": Object {},
                      "setup": [Function],
                    },
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray"
                >
                  <h1
                    class="text-6xl font-black text-brand-gray-dark"
                  >
                    Coloque seu Widget bem aqui
                  </h1>
                  <widget-stub />
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "w-full h-3/4 flex flex-col justify-center items-center bg-brand-gray",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": Object {
                  "a": null,
                  "accessCache": Object {
                    "@@__IMMUTABLE_ITERABLE__@@": 0,
                    "@@__IMMUTABLE_RECORD__@@": 0,
                    "_isMockFunction": 0,
                    "constructor": 0,
                    "isVueInstance": 0,
                    "toJSON": 0,
                    Symbol(Symbol.toStringTag): 0,
                  },
                  "appContext": Object {
                    "app": [Circular],
                    "components": Object {
                      "transition": Object {
                        "name": "transition",
                      },
                      "transition-group": Object {
                        "name": "transition-group",
                      },
                    },
                    "config": Object {
                      "compilerOptions": Object {},
                      "errorHandler": undefined,
                      "globalProperties": Object {},
                      "isNativeTag": [Function],
                      "optionMergeStrategies": Object {},
                      "performance": false,
                      "warnHandler": undefined,
                    },
                    "directives": Object {},
                    "emitsCache": WeakMap {},
                    "mixins": Array [
                      Object {
                        "beforeCreate": [Function],
                      },
                    ],
                    "optionsCache": WeakMap {},
                    "propsCache": WeakMap {},
                    "provides": Object {},
                    "reload": [Function],
                  },
                  "asyncDep": null,
                  "asyncResolved": false,
                  "attrs": Object {},
                  "attrsProxy": Object {},
                  "bc": null,
                  "bm": null,
                  "bu": null,
                  "bum": null,
                  "c": null,
                  "components": null,
                  "ctx": Object {},
                  "da": null,
                  "data": Object {},
                  "directives": null,
                  "ec": null,
                  "effect": ReactiveEffect {
                    "active": true,
                    "allowRecurse": true,
                    "deps": Array [
                      Set {
                        [Circular],
                      },
                    ],
                    "fn": [Function],
                    "onTrack": undefined,
                    "onTrigger": undefined,
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                  "emit": [Function],
                  "emitsOptions": null,
                  "emitted": null,
                  "exposeProxy": null,
                  "exposed": null,
                  "inheritAttrs": undefined,
                  "isDeactivated": false,
                  "isMounted": true,
                  "isUnmounted": false,
                  "m": null,
                  "next": null,
                  "parent": [Circular],
                  "props": Object {},
                  "propsDefaults": Object {},
                  "propsOptions": Array [
                    Object {},
                    Array [],
                  ],
                  "provides": Object {},
                  "proxy": Object {},
                  "refs": Object {},
                  "render": [Function],
                  "renderCache": Array [],
                  "root": [Circular],
                  "rtc": null,
                  "rtg": null,
                  "scope": EffectScope {
                    "_active": true,
                    "cleanups": Array [],
                    "detached": true,
                    "effects": Array [
                      ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": Array [
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                    ],
                    "parent": undefined,
                  },
                  "setupContext": Object {
                    "attrs": Object {},
                    "emit": [Function],
                    "expose": [Function],
                    "slots": Object {},
                  },
                  "setupState": Object {},
                  "slots": Object {},
                  "slotsProxy": Object {},
                  "sp": null,
                  "subTree": Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <widget-stub />,
                    "key": null,
                    "patchFlag": 0,
                    "props": Object {},
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "widget-stub",
                  },
                  "suspense": null,
                  "suspenseId": 0,
                  "type": Object {
                    "model": undefined,
                    "name": "Widget",
                    "props": Object {},
                    "setup": [Function],
                  },
                  "u": null,
                  "uid": 2,
                  "um": null,
                  "update": [Function],
                  "vnode": [Circular],
                  "withProxy": null,
                },
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <widget-stub />,
                "key": null,
                "patchFlag": 0,
                "props": null,
                "ref": null,
                "scopeId": null,
                "shapeFlag": 4,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Object {
                  "model": undefined,
                  "name": "Widget",
                  "props": Object {},
                  "setup": [Function],
                },
              },
            ],
            "dynamicProps": null,
            "el": ,
            "key": null,
            "patchFlag": 64,
            "props": null,
            "ref": null,
            "scopeId": null,
            "shapeFlag": 16,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": Symbol(v-fgt),
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "__name": "PlaygroundIndex",
            "components": Object {},
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "__name": "PlaygroundIndex",
          "components": Object {},
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [],
  "componentVM": Object {},
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {},
  },
  "wrapperElement": ,
}
`;
